### PR TITLE
Implementation of Permute CL kernel to handle all permutations

### PR DESCRIPTION
**Pull request by shub98-gupta**
This patch will add a generic permute cl-kernel to handle
all permutations available for tensors having rank upto 4.

Signed-off-by: shubham <shub98.gupta@samsung.com>

### CODE REVIEW

**Comment by shub98-gupta** [comment]
Related issue #556

File: src/core/CL/kernels/CLPermuteKernel.cpp [hunk 1]
```diff
@@ -56,10 +56,6 @@ Status validate_arguments(const ITensorInfo *input, const ITensorInfo *output, c
                                                          DataType::U16, DataType::S16,
                                                          DataType::U32, DataType::S32,
                                                          DataType::F16, DataType::F32);
-    ARM_COMPUTE_RETURN_ERROR_ON_MSG((perm != PermutationVector{ 2U, 0U, 1U })
```
Comment (GeorgeARM) [inline src/core/CL/kernels/CLPermuteKernel.cpp:59]: You should return error if the permutation vector size or the tensor dimensions are greater than 4

**Comment by GeorgeARM** [comment]
One more thing @shub98-gupta,

You should submit your patch in https://review.mlplatform.org/#/admin/projects/ml/ComputeLibrary
All our development has been moved to public there, github is maintained only for releases

File: src/core/CL/kernels/CLPermuteKernel.cpp (hunk 1 above)
Comment (shub98-gupta) [reply src/core/CL/kernels/CLPermuteKernel.cpp:59]: Okay sure

**Comment by shub98-gupta** [comment]
@GeorgeARM, okay I will submit this patch to the specified link.

**Comment by shub98-gupta** [comment]
@GeorgeARM whenever i am trying to push to the gerrit link as mentioned by you. I am getting the following error.

E:\git\ComputeLibrary>git push origin HEAD:refs/for/master
remote: Unauthorized
fatal: Authentication failed for 'https://review.mlplatform.org/ml/ComputeLibrary/'

I have cloned the repo by using git clone https://review.mlplatform.org/ml/ComputeLibrary

Can u pls help me to resolve this prob?

**Comment by AnthonyARM** [comment]
You need to clone the repo using ssh not http and you need to set up a ssh key on your account on that server

**Comment by shub98-gupta** [comment]
Hi @AnthonyARM @GeorgeARM 
I have uploaded the patch but ArmJenkins is producing this error, Can u pls tell what is it and how to resolve ?

SHUBHAM GUPTA
Uploaded patch set 1.1:04 PM
Arm Jenkins
1:04 PM
↩
Patch Set 1:

Build Started http://blg-compute.cambridge.arm.com/jenkins/job/ComputeLibraryLinaro_precommit/861/
Arm Jenkins
Patch Set 1: ERROR: Email address shub98.gupta@samsung.com not on the `whitelist`, ask for a maintainer to manually trigger your job1:04 PM
Arm Jenkins
Patch Set 1: Verified-1 Build Failed http://blg-compute.cambridge.arm.com/jenkins/job/ComputeLibraryLinaro_precommit/861/ : FAILURE1:04 PM

**Comment by AnthonyARM** [comment]
Hi @shub98-gupta ,
It's normal, in order to avoid DoS attacks we've set up a whitelist for people allowed to submit patches.
I've just manually triggered the Jenkins job for it

**Comment by GeorgeARM** [comment]
This has been merged on top of the development master
https://git.mlplatform.org/ml/ComputeLibrary.git/commit/?h=refs/heads/master&id=e1a4e37ac266324de366401f2bda459b011a4869

**Pull request closed without merging**